File: src/java_sub/242.有效的字母异位词.java
import java.util.Arrays;

/*
 * @lc app=leetcode.cn id=242 lang=java
 *
 * [242] 有效的字母异位词
 * 将两个字符串转成字符数组然后排序，排序后比较两个数组是否完全一样；
 * 注意的是用Arrays.equals比较字符数组的值
 */

// @lc code=start
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length())
            return false;

        char[] sArray = s.toCharArray();
        char[] tArray = t.toCharArray();

        Arrays.sort(sArray);
        Arrays.sort(tArray);

        return Arrays.equals(sArray, tArray);

    }
}
// @lc code=end

File: src/py3/排序和字符串/242.有效的字母异位词.py
#
# @lc app=leetcode.cn id=242 lang=python
#
# [242] 有效的字母异位词
#

# @lc code=start
class Solution(object):


    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False

        counter = {chr(97+i):0 for i in range(26)}
        for i in range(len(s)):
            counter[s[i]] += 1
            counter[t[i]] -= 1

        tmp = [False for i in range(26) if counter[chr(97+i)] != 0]
        return len(tmp) == 0

    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False

        counter = {}
        for i in range(len(s)):
            counter[s[i]] = counter.setdefault(s[i], 0) + 1
            counter[t[i]] = counter.setdefault(t[i], 0) - 1

        tmp = [False for i in counter if counter[i] != 0]


        return len(tmp) == 0


    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False

        counter = {}
        for i in range(len(s)):
            counter[s[i]] = counter.setdefault(s[i], 0) + 1
            counter[t[i]] = counter.setdefault(t[i], 0) - 1

        return not any(counter[i] != 0 for i in counter)




# @lc code=end

